File: lib/java/main/src/main/java/alphaTab/alphaSkia/AlphaSkiaData.java
package alphaTab.alphaSkia;

class AlphaSkiaData extends AlphaSkiaNative {
    AlphaSkiaData(long handle) {
        super(handle);
    }

    AlphaSkiaData(byte[] raw) {
        super(allocateCopy(raw));
    }

    private static native long allocateCopy(byte[] raw);

    @Override
    public native void close();

    public native byte[] toArray();
}

File: lib/kotlin/mpp/src/commonMain/kotlin/alphaTab/alphaSkia/AlphaSkiaData.kt
package alphaTab.alphaSkia

internal class AlphaSkiaData(raw: ByteArray) :
    AlphaSkiaNative(NativeMethods.alphaskiaDataNewCopy(raw)) {

    override fun close() {
        NativeMethods.alphaskiaDataFree(this.handle)
    }

    fun toArray(): ByteArray {
        return NativeMethods.alphaskiaDataGetData(this.handle)
    }
}
